pkg/user/tx_client.go

```diff
@@ -137,6 +148,9 @@ type TxClient struct {
 	defaultGasPrice float64
 	defaultAccount  string
 	defaultAddress  sdktypes.AccAddress
+	txPool          map[string]poolTxInfo
```

Comment:
[Question] do we want to have a way of periodically clearing the cache?

Reply:
@evan-forbes @cmwaters I'd appreciate your input here

Reply:
ideally, it seems like we should never be a scenario where we need this to be cleared. Either a tx is included and we continually try to get it included, or we handle each case when that does not occur and remove the tx at that point

Reply:
the only time tx doesn't get removed from this local cache is if context gets cancelled while polling for tx status which means that the said tx was never confirmed. If the transaction doesn't get confirmed it can linger in that cache until it ooms

Reply:
by periodically? do we mean a ttl? imo, the most important thing here is to be able to consistently reason about what is occurring in the local pool.